analytics-integrations/google-analytics/src/main/java/com/segment/analytics/internal/integrations/GoogleAnalyticsIntegration.java

```diff
@@ -166,6 +202,20 @@ boolean handleProductEvent(String event, String category, Properties properties)
     return false;
   }
 
+  void setCustomDimensions(HitBuilders.HitBuilder hit, Properties properties) {
```

Comment:
We could create two new classes `EventHitBuilder` and `AppHitViewBuilder` that extend `EventBuilder` and `AppViewBuilder` respectively, and implement a new interface `CustomHitBuilder` that this method accepts. 

`void setCustomDimensions(CustomHitBuilder hit, Properties properties)`

```java
  interface CustomHitBuilder {
    CustomHitBuilder setCustomDimension(int index, String dimension);
  }

  static class EventHitBuilder extends HitBuilders.EventBuilder implements CustomHitBuilder {
    @Override public EventHitBuilder setCustomDimension(int index, String dimension) {
      super.setCustomDimension(index, dimension);
      return this;
    }
  }

  static class AppViewHitBuilder extends HitBuilders.EventBuilder implements CustomHitBuilder {
    @Override public AppViewHitBuilder setCustomDimension(int index, String dimension) {
      super.setCustomDimension(index, dimension);
      return this;
    }
  }
```